src/app/(admin)/admin/_lib/validationSchemas/specialistCommonSchemas.js

```diff
@@ -35,6 +36,50 @@ export const zInteger = z
 
 export const zUrl = zString.url({ message: MESSAGES.unacceptableValue });
 
+export const zWorkTimeSchema = z
+  .array(
+    z
+      .object({
+        weekDay: z.enum(Object.values(WEEKDAYS_TRANSLATION)),
+        time: z
+          .string()
+          .refine(val => !val || /\d{2}:\d{2}\s-\s\d{2}:\d{2}/.test(val), {
+            message: 'Введіть час у форматі ХХ:ХХ - ХХ:ХХ',
+          })
+          .nullish(),
+        isDayOff: z.boolean().nullish(),
+      })
+      .superRefine((data, ctx) => {
+        const { time, isDayOff } = data;
+        if (time && isDayOff) {
+          ctx.addIssue({
+            code: 'custom',
+            message: 'Приберіть час роботи, якщо це вихідний',
+            path: ['time'],
+          });
+        }
+      }),
+  )
+  .superRefine((workTime, ctx) => {
+    const specified = isSpecifiedWorkTime(workTime);
+    if (specified) {
```

Comment:
It's better to use guard-statement as you did before:

```javascript
    if (!isSpecifiedWorkTime(workTime)) return;
    // Following code without nested "if"
```

Reply:
Changed it